pkg/istio-agent/agent.go

```diff
@@ -406,13 +406,30 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 		return nil, fmt.Errorf("failed to start local DNS server: %v", err)
 	}
 
-	a.secretCache, err = a.newSecretManager()
-	if err != nil {
-		return nil, fmt.Errorf("failed to start workload secret manager %v", err)
-	}
+	if socketFileExists(security.WorkloadIdentitySocketPath) {
+		log.Info("SDS socket found. Istio SDS Server won't be started")
+	} else {
```

Comment:
nit (I'm not really doing code maintenance anymore so I defer to the folks that actually are these days): This func is getting pretty unwieldy - I'd say pull the `else` out into an `a.initiateSDS()` func.

Reply:
Thank you @ZackButcher, that's a good suggestion, I did that refactor as part of my latest changes where I also added the socket health check.